source/jquery.flot.js

```diff
@@ -2496,16 +2496,31 @@ Licensed under the MIT license.
 
         function findNearbyItems(mouseX, mouseY, seriesFilter, radius, computeDistance) {
             var items = findItems(mouseX, mouseY, seriesFilter, radius, computeDistance);
-            executeHooks(hooks.findNearbyItems, [mouseX, mouseY, seriesFilter, radius, computeDistance, items]);
-            return items;
+            for (var i = 0; i < series.length; ++i) {
+                if (seriesFilter(i)) {
+                    executeHooks(hooks.findNearbyItems, [mouseX, mouseY, series, i, radius, computeDistance, items]);
+                }
+            }
+
+            return items.sort((a, b) => { 
+                if (b.distance === undefined) {
+                    return -1;
+                } else if (a.distance === undefined && b.distance !== undefined) {
+                    return 1;
+                }
+
+                return a.distance - b.distance ;
+            });
         }
 
         function findNearbyItem(mouseX, mouseY, seriesFilter, radius, computeDistance) {
             var items = findNearbyItems(mouseX, mouseY, seriesFilter, radius, computeDistance);
             var closest = items[0];
  
             for (var i = 1; i < items.length; ++i) {
-               if (closest.distance != undefined && items[i].distance < closest.distance) {
+               if (closest.distance !== undefined && 
```

Comment:
We just need to return the first item in the array now, since we're sorting in findNearbyItems right?

Reply:
Yep!